.github/workflows/tests.yaml

```diff
@@ -40,14 +40,70 @@ jobs:
     - name: run unit tests
       run: make test.unit
 
-    - name: run integration tests
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v3
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        fail_ci_if_error: true
+        files: unit.coverage.out
+        verbose: true
+
+  setup-integration-tests:
+    runs-on: ubuntu-latest
+    outputs:
+      test_names: ${{ steps.set_test_names.outputs.test_names }}
+    steps:
+    - uses: actions/checkout@v3
+    - id: set_test_names
+      name: Set test names
+      working-directory: test/integration/
+      # grep magic described in https://unix.stackexchange.com/a/13472
+      # sed to add the extra $ is because some of our test names overlap. we need it so the -run regex only matches one test
+      run: |
+        echo "::set-output name=test_names::$(grep -shoP "(?<=^func )(Test[a-zA-z_0-9]+)(?=\(t \*testing.T\) {)" * | sed -e "s/$/\$/"| jq -R . | jq -cs .)"
```

Comment:
Knowing that `set-output` is to be deprecated, can we make it already use env file?

Reply:
You're right. I was just lazy I guess. PTAL now.